### PR TITLE
[PLAYER-5615] ContentProtectionSampleApp updated

### PR DESCRIPTION
**Pull request by maxkupetskii**
- uses `OptionsDataSource`
- uses `BasicEmbedTokenGenerator`

### CODE REVIEW

**Review by cacel — APPROVED** [review]

File: ContentProtectionSampleApp/ContentProtectionSampleApp/Lists/ContentProtectionListViewController.m [hunk 1]
```diff
@@ -32,56 +29,7 @@ - (void)viewDidLoad {
...
-//                                                 domain:@"http://www.ooyala.com"
-//                                         viewController:DeviceManagementPlayerViewController.class],
-//  TODO: [PBA-5945] Fix asset configuration
-//           [[PlayerSelectionOption alloc] initWithTitle:@"FairPlay (Unconfigured)"
-//                                              embedCode:@"BuMjEwMDE6b8-bX45pBkcgFieNehCcln"
-//                                                  pcode:@"RkcjMxOtMYDwJzrPy3sWJLl6blS1"
-//                                                 domain:@"http://www.ooyala.com"
-//                                         viewController:FairplayPlayerViewController.class]
-           ];
+  _options = OptionsDataSource.options;
```
Comment (oSulyma) [inline ContentProtectionSampleApp/ContentProtectionSampleApp/Lists/ContentProtectionListViewController.m]: Please use `self.` according to our code style.
Is it intended that all three properties are atomic? ([By default, an Objective-C property is atomic](https://developer.apple.com/library/archive/documentation/Cocoa/Conceptual/ProgrammingWithObjectiveC/EncapsulatingData/EncapsulatingData.html#//apple_ref/doc/uid/TP40011210-CH5-SW2))

File: ContentProtectionSampleApp/ContentProtectionSampleApp/Players/DeviceManagementPlayerViewController.m [hunk 2]
```diff
@@ -54,6 +48,23 @@ - (instancetype)initWithPlayerSelectionOption:(PlayerSelectionOption *)playerSel
...
+    self.apiKey        = tokenGenerator.apiKey;
+    self.secret        = tokenGenerator.apiSecret;
+    self.accountId     = tokenGenerator.accountId;
+    self.authorizeHost = tokenGenerator.authorizeHost;
+  } else {
+    self.apiKey        = @"API_KEY";
+    self.secret        = @"API_SECRET";
+    self.accountId     = @"ACCOUNT_ID";
+    self.authorizeHost = @"AUTHORIZE_HOST";
+  }
```
Comment (oSulyma) [inline ContentProtectionSampleApp/ContentProtectionSampleApp/Players/DeviceManagementPlayerViewController.m:67]: Please, use iVars above

File: ContentProtectionSampleApp/ContentProtectionSampleApp/Players/DeviceManagementPlayerViewController.m [hunk 3]
```diff
@@ -54,6 +48,23 @@ - (instancetype)initWithPlayerSelectionOption:(PlayerSelectionOption *)playerSel
     NSLog(@"There was no PlayerSelectionOption!");
     return nil;
   }
+  /*
+   * The API Key and Secret should not be saved inside your applciation (even in git!).
```
Comment (oSulyma) [inline ContentProtectionSampleApp/ContentProtectionSampleApp/Players/DeviceManagementPlayerViewController.m]: Typo in word **applciation**

File: ContentProtectionSampleApp/ContentProtectionSampleApp/Players/DeviceManagementPlayerViewController.m [hunk 4]
```diff
@@ -54,6 +48,23 @@ - (instancetype)initWithPlayerSelectionOption:(PlayerSelectionOption *)playerSel
...
+   * However, for debugging you can use them to locally generate Ooyala Player Tokens.
+   */
+  if (self.playerSelectionOption.embedTokenGenerator &&
+      [self.playerSelectionOption.embedTokenGenerator isKindOfClass:BasicEmbedTokenGenerator.class]) {
+    BasicEmbedTokenGenerator *tokenGenerator = (BasicEmbedTokenGenerator *)self.playerSelectionOption.embedTokenGenerator;
+    self.apiKey        = tokenGenerator.apiKey;
+    self.secret        = tokenGenerator.apiSecret;
+    self.accountId     = tokenGenerator.accountId;
+    self.authorizeHost = tokenGenerator.authorizeHost;
```
Comment (oSulyma) [inline ContentProtectionSampleApp/ContentProtectionSampleApp/Players/DeviceManagementPlayerViewController.m]: Why most of properties above don't have `nonatomic` attribute? Taking in account that getter and setter for `atomic` are more expensive methods, I tends not to use it without reasoning.


.

File: ContentProtectionSampleApp/ContentProtectionSampleApp/Players/FairplayPlayerViewController.m [hunk 5]
```diff
@@ -55,10 +64,19 @@ - (instancetype)initWithPlayerSelectionOption:(PlayerSelectionOption *)playerSel
    * The API Key and Secret should not be saved inside your applciation (even in git!).
```
Comment (oSulyma) [inline ContentProtectionSampleApp/ContentProtectionSampleApp/Players/FairplayPlayerViewController.m]: Typo in **applciation**

File: ContentProtectionSampleApp/ContentProtectionSampleApp/Players/DeviceManagementPlayerViewController.m [hunk 6]
```diff
@@ -1,15 +1,14 @@
...
+//  Created on 5/15/12.
+//  Copyright © 2012 Ooyala Inc. All rights reserved.
+//
 
 #import "DeviceManagementPlayerViewController.h"
-#import "AdobePassViewController.h"
 #import <OoyalaSDK/OoyalaSDK.h>
+#import "BasicEmbedTokenGenerator.h"
 
 @interface DeviceManagementPlayerViewController () <OOEmbedTokenGenerator>
```
Comment (oSulyma) [inline ContentProtectionSampleApp/ContentProtectionSampleApp/Players/DeviceManagementPlayerViewController.m:13]: Please add #pragma marks, at least for implementation `OOEmbedTokenGenerator ` protocol

**Review by oSulyma — CHANGES_REQUESTED** [review]
I added six comments/remarks.

File: ContentProtectionSampleApp/ContentProtectionSampleApp/Lists/ContentProtectionListViewController.m (hunk 1 above)
Comment (maxkupetskii) [reply ContentProtectionSampleApp/ContentProtectionSampleApp/Lists/ContentProtectionListViewController.m]: 1. `viewDidLoad` can be treated as `init`
2. I did not change those properties and their characteristics.

File: ContentProtectionSampleApp/ContentProtectionSampleApp/Players/DeviceManagementPlayerViewController.m (hunk 6 above)
Comment (maxkupetskii) [reply ContentProtectionSampleApp/ContentProtectionSampleApp/Players/DeviceManagementPlayerViewController.m:13]: What for? It's purely optional thing

Comment (oSulyma) [reply ContentProtectionSampleApp/ContentProtectionSampleApp/Players/DeviceManagementPlayerViewController.m:13]: More useful. Agree, optional.

**Pull request merged**